blueocean-dashboard/src/main/js/components/testing/TestService.js

```diff
@@ -47,17 +59,12 @@ export default class TestService extends BunkerService {
         return this._logs;
     }
 
-    static createURL({ run, status, state }) {
-        if (status && state) {
-            throw new Error('status and state are exclusive');
-        }
+    static createURL({ run, status, state, age }) {
```

Comment:
This isn't exported, so I'm not sure why exactly the signature is what it is. I'd change the signature to this:

`static createURL(run, searchParams)`

Then explicitly check `searchParams` for "at least one" if that check is really needed, and then just hand that object off to `new URLSearchParams()`. Make the calling functions stop putting null props into it in the first place.

Reply:
Agreed, I like your idea way better, so done. Way cleaner